src/utils/processInputs/index.ts

```diff
@@ -0,0 +1,37 @@
+import { FormattedAbiParameter } from '../../types/parameter'
+import { Extras } from '../../types/base'
+import parse from './parse'
+import { RequiredAllowances } from '../../types'
+import getTagCallback, { GetTagCallbackParams } from './getTagCallback'
+
+export type ParseInputsBaseParams = {
+  extras?: Extras
+  formattedInputs: readonly FormattedAbiParameter[]
+  args: any
+}
+
+export type ParseInputsParams = Omit<GetTagCallbackParams, 'requiredAllowances'>
+
+export default async function processInputs(params: ParseInputsParams) {
+  const requiredAllowances: RequiredAllowances[] = []
```

Comment:
Am I understanding correctly that the callback takes `requiredAllowances` and actually modfies it? If that is the case I would find it clearer if `requiredAllowances` would be `let` and re-assigned via spread operator or so. Not a necessity, but clearer imo.

Reply:
yeah it is mutating it from the child process

Reply:
im not sure if you can set it with a spread since its updating it on every for loop cycle